### PR TITLE
Fix segmentation fault on vector reallocation

### PR DESCRIPTION
**Pull request by hansbinderup**
Fixes double allocation for vector as element was inserted before deleted.
Also fixes dangling iterators when vector reallocates.

Would be better with a fixed sized circle buffer - will be for another day. 
`std::list` should be alright for memory constrained devices as memory is no longer stored contiguously.

### CODE REVIEW

File: include/Yash.h [hunk 1]
```diff
@@ -394,11 +393,11 @@ class Yash {
...
     std::array<std::string_view, TConfig.maxRequiredArgs> m_commandArgs;
     std::function<void(const char*)> m_printFunction;
-    std::vector<std::string> m_commandHistory;
-    std::vector<std::string>::const_iterator m_commandHistoryIndex;
-    std::string m_command;
-    std::string m_prompt;
-    std::string m_ctrlCharacter;
+    std::list<std::string> m_commandHistory;
+    std::list<std::string>::const_iterator m_commandHistoryIndex;
+    std::string m_command { "" };
```
Comment (mads-bn) [inline include/Yash.h]: std::string is already by default an empty string - right?

File: include/Yash.h [hunk 2]
```diff
@@ -394,11 +393,11 @@ class Yash {
...
+    std::string m_prompt { "Yash$ " };
```
Comment (mads-bn) [inline include/Yash.h:403]: "Yash$ " why this change?

**Review by mads-bn — APPROVED** [review]

File: include/Yash.h (hunk 1 above)
Comment (hansbinderup) [reply include/Yash.h:398]: Woops, will also added reservation of the command in next PR so it's actually not relevant here.. Will remove here :+1:

File: include/Yash.h (hunk 2 above)
Comment (hansbinderup) [reply include/Yash.h:403]: Empty prompt looks weird and I thought that it was a nice gimmick :)

File: include/Yash.h (hunk 1 above)
Comment (hansbinderup) [reply include/Yash.h]: Yep, was just overly explicit. I'll remove :+1:

File: include/Yash.h [hunk 3]
```diff
@@ -74,11 +74,10 @@ class Yash {
             print("\r\n");
             if (!m_command.empty()) {
                 runCommand();
-                m_commandHistory.push_back(m_command);
-
-                if (m_commandHistory.size() > m_commandHistorySize)
+                if (m_commandHistory.size() >= m_commandHistorySize)
```
Comment (Emil-Juhl) [inline include/Yash.h]: Probably should handle the case of size == 0 here :) Seems like it's allowed to not have a history

**Review by Emil-Juhl — CHANGES_REQUESTED** [review]

File: include/Yash.h (hunk 3 above)
Comment (hansbinderup) [reply include/Yash.h]: Good idea :+1:

**Review by Emil-Juhl — APPROVED** [review]
